### DISCUSSION TITLE
Scheduler long running tasks

### DISCUSSION THREAD

**Discussion by chapterjason** [symfony/symfony#52278] [Q&A] [answered]
I have some tasks that might take longer then the interval the message will be rescheduled again.
I created a workaround by wrapping my messages with an `AsyncMessage` like that:

```php
final class AsyncMessage
{
    private mixed $message;

    public function __construct(mixed $message)
    {
        $this->message = $message;
    }

    public function getMessage(): mixed
    {
        return $this->message;
    }
}
```

```php
#[AsMessageHandler]
final class AsyncMessageHandler
{
    private MessageBusInterface $messageBus;

    public function __construct(MessageBusInterface $messageBus)
    {
        $this->messageBus = $messageBus;
    }

    public function __invoke(AsyncMessage $message)
    {
        $this->messageBus->dispatch($message->getMessage());
    }
}
```

I just wondered, why is the scheduler also handling the messages and not just dispatch them?

**Comment by stof** [comment, marked as answer]
The scheduler handles messages because this is the efficient way to use resources when processing is faster than the interval (instead of staying there waiting until the end of the interval) and because it also avoids issues where your messenger consumer has a long queue and your dispatched message gets processed only later.

However, the case of messages that should be dispatched in the queue is a valid one. And the good news is that this is supported out of the box. Your own solution even matches the way it has been implemented in core (except the naming). Make your schedule use `Symfony\Component\Messenger\Message\RedispatchMessage` as its message.

**Reply by chapterjason** [reply]
Thanks for clarification and also for the insight of the `RedispatchMessage`, didn't know about that one!
